### PR TITLE
go-task: fix shell completion

### PR DESCRIPTION
**Pull request by Zebradil**
## Description of changes

This PR adds code to the `postInstall` hook to patch the zsh completions file to provide completions for the `go-task` command.
Without this change, completion works only for the `task` command but not for `go-task`.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by SigmaSquadron** [comment]
Welcome to Nixpkgs! Please rebase your PR to `master` instead of `nixos-unstable`. Development happens on the `master` and `staging` branches, and only Hydra, the CI build system, updates the `nixos-*` branches, which is then used by end users.

**Comment by Aleksanaa** [comment]
What about bash and fish? Do they have the same problem?

**Comment by Zebradil** [comment]
Sorry for the late follow up. I was struggling to get testing environment configured to test in all shells.



> What about bash and fish? Do they have the same problem?

@Aleksanaa yes, you were right, the same issue was in fish and bash.

**Comment by smancill** [comment]
Please edit the second commit with `go-task: fix bash and fish completion`.

Or even better, rebase and squash them both.

**Comment by Zebradil** [comment]
Done, now it's a single commit [go-task: fix shell completion for command alias](https://github.com/NixOS/nixpkgs/pull/332696/commits/099f709d9ec2024222db5f33bdc3e08ef1ebfad2)

**Review by smancill — APPROVED** [review]
LGTM

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 332696` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>go-task</li>
  </ul>
</details>

File: pkgs/development/tools/go-task/default.nix [hunk 1]
```diff
@@ -46,6 +46,13 @@ buildGoModule rec {
     ln -s $out/bin/task $out/bin/go-task
 
     installShellCompletion completion/{bash,fish,zsh}/*
+
+    substituteInPlace $out/share/bash-completion/completions/task.bash \
+      --replace-warn 'complete -F _task task' 'complete -F _task task go-task'
+    substituteInPlace $out/share/fish/vendor_completions.d/task.fish \
+      --replace-warn 'complete -c $GO_TASK_PROGNAME' 'complete -c $GO_TASK_PROGNAME -c go-task'
+    substituteInPlace $out/share/zsh/site-functions/_task \
+      --replace-warn '#compdef task' '#compdef task go-task'
```
Comment (SuperSandro2000) [inline pkgs/development/tools/go-task/default.nix:55]: This should be using --replace-fail to fail the build if it mismatches.

Comment (Zebradil) [reply pkgs/development/tools/go-task/default.nix:55]: Fine for me. Here you go: #339010

**Pull request merged**